src/main/extension/extension_install.cpp

```diff
@@ -88,6 +88,9 @@ string ExtensionHelper::ExtensionDirectory(DatabaseInstance &db, FileSystem &fs)
 		// expand ~ in extension directory
 		extension_directory = fs.ExpandPath(extension_directory);
 		if (!fs.DirectoryExists(extension_directory)) {
+			if (!create) {
+				return extension_directory;
+			}
```

Comment:
There is a corner case where `extension_directory` exists, and the subdirectories are still created.

Possibly create should wrap all calls of `CreateDirectory` (2 locations).